arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -711,9 +711,27 @@ public void traceContextExit(MessageFrame frame) {
   private void exitDeploymentFromDeploymentInfoPointOfView() {
 
     // sanity check
+    if (this.state.processingPhase != TX_SKIP) {
+      Preconditions.checkArgument(
+          deploymentStatusOfBytecodeAddress()
+              == (messageFrame().getType() == MessageFrame.Type.CONTRACT_CREATION),
```

Comment:
.equals ?

Reply:
It's an `enum` comparison, no `.equals(...)` required.